### PR TITLE
Migrate AWS resources (aws_vpc_endpoint, aws_route53_resolver_endpoint, aws_route53_record)

### PR DESCRIPTION
**Pull request by aliscott**
As part of https://github.com/infracost/infracost/issues/1288

### CODE REVIEW

**Comment by aliscott** [comment]
@sinabakh can you review this please?

**Comment by sinabakh** [comment]
Sure, I will do it today.

File: internal/providers/terraform/aws/route53_record.go [hunk 1]
```diff
@@ -1,107 +1,25 @@
...
-		}
-
-		if geoQueriesTiers[1].GreaterThan(decimal.Zero) {
-			costComponents = append(costComponents, queriesCostComponent("Geo DNS queries (over 1B)", "Geo-Queries", "1000000000", &geoQueriesTiers[1]))
-		}
-	} else {
-		var unknown *decimal.Decimal
-
-		costComponents = append(costComponents, queriesCostComponent("Geo DNS queries (first 1B)", "Geo-Queries", "0", unknown))
+	if len(d.References("alias.0.name")) > 0 && d.References("alias.0.name")[0].Type != "aws_route53_record" {
```
Comment (sinabakh) [inline internal/providers/terraform/aws/route53_record.go:18]: Although this is ok from a migration point of view but could you please explain to me what comparing a `Type` to a string mean?

File: internal/providers/terraform/aws/route53_resolver_endpoint.go [hunk 2]
```diff
@@ -1,86 +1,20 @@
...
-	if u != nil && u.Get("monthly_queries").Exists() {
-		monthlyQueries := decimal.NewFromInt(u.Get("monthly_queries").Int())
-		dnsQueriesTier := usage.CalculateTierBuckets(monthlyQueries, queryTierLimits)
-		tierOne := dnsQueriesTier[0]
-		tierTwo := dnsQueriesTier[1]
-
-		if tierOne.GreaterThan(decimal.NewFromInt(0)) {
-			costComponents = append(costComponents, dnsQueriesCostComponent(region, "DNS queries (first 1B)", "0", &tierOne))
-		}
+	resolverEndpoints := int64(len(d.Get("ip_address").Array()))
```
Comment (sinabakh) [inline internal/providers/terraform/aws/route53_resolver_endpoint.go:15]: How about we rename it to `resolverEndpointsLen`?

File: internal/providers/terraform/aws/vpc_endpoint.go [hunk 3]
```diff
@@ -1,128 +1,28 @@
...
-			IsSkipped: true,
-		}
-	}
+	subnetIds := d.Get("subnet_ids").Array()
 
-	if u != nil && u.Get("monthly_data_processed_gb").Exists() {
-		gbDataProcessed = decimalPtr(decimal.NewFromFloat(u.Get("monthly_data_processed_gb").Float()))
+	vpcEndpointInterfaces := int64(1)
+	if len(subnetIds) > 0 {
+		vpcEndpointInterfaces = int64(len(subnetIds))
```
Comment (sinabakh) [inline internal/providers/terraform/aws/vpc_endpoint.go:19]: change to `vpcEndpointInterfacesLen`?

File: internal/providers/terraform/aws/route53_record.go (hunk 1 above)
Comment (aliscott) [reply internal/providers/terraform/aws/route53_record.go:18]: `.Type` on resource_data is a string that stores the Terraform resource type (https://github.com/infracost/infracost/blob/master/internal/schema/resource_data.go#L12)

File: internal/providers/terraform/aws/route53_resolver_endpoint.go (hunk 2 above)
Comment (aliscott) [reply internal/providers/terraform/aws/route53_resolver_endpoint.go]: I've updated to `resolverEndpointsCount` but still kept the struct property as `ResolverEndpoints` so it's consistent with what we do for (most) other integer properties and in the usage file.

File: internal/providers/terraform/aws/vpc_endpoint.go (hunk 3 above)
Comment (aliscott) [reply internal/providers/terraform/aws/vpc_endpoint.go]: As above, have updated to `vpcEndpointInterfaceCount`

**Comment by sinabakh** [comment]
LGTM

**Pull request merged**